llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -374,6 +374,8 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
       .clampScalar(0, s32, sXLen)
       .lowerForCartesianProduct({s32, sXLen, p0}, {p0});
 
+  getActionDefinitionsBuilder(G_VSCALE).customFor({sXLen});
```

Comment:
Don't you need a clamp scalar if the type isn't sXLen? The vectorizer always uses s32.

Reply:
I can add that. This patch is a precommit for G_INSERT and G_EXTRACT which only build sXLen G_VSCALE instructions. So it was not needed for that patch.

Reply:
Added